File: feign-core/src/main/java/name/abhijitsarkar/feign/core/model/DefaultIdGenerator.java
package name.abhijitsarkar.feign.core.model;

import name.abhijitsarkar.feign.core.domain.Request;

import java.util.UUID;

/**
 * @author Abhijit Sarkar
 */
public class DefaultIdGenerator implements IdGenerator {
    @Override
    public String id(Request request) {
        return UUID.randomUUID().toString();
    }
}

File: app/org/abhijitsarkar/feign/service/DefaultIdGenerator.scala
package org.abhijitsarkar.feign.service

import org.abhijitsarkar.feign.api.model.Request
import org.abhijitsarkar.feign.api.persistence.IdGenerator

/**
  * @author Abhijit Sarkar
  */
final class DefaultIdGenerator extends IdGenerator {
  val pattern = """^(?:/?)([^/]+)(?:.*)$""".r

  override def id(request: Request): String = {
    val prefix = pattern.findFirstMatchIn(request.path)
      .map(_ group 1)
      .getOrElse("unknown")

    s"${prefix}-${request.path.hashCode}"
  }
}
